internal/git/util.go

```diff
@@ -0,0 +1,201 @@
+package git
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+
+	"github.com/go-git/go-git/v5"
+	"github.com/go-git/go-git/v5/config"
+	"github.com/go-git/go-git/v5/plumbing"
+	"github.com/go-git/go-git/v5/plumbing/object"
+	"github.com/numaproj-labs/numaplane/api/v1alpha1"
+	"github.com/numaproj-labs/numaplane/internal/kustomize"
+	"github.com/numaproj-labs/numaplane/internal/shared/logging"
+	k8sClient "sigs.k8s.io/controller-runtime/pkg/client"
+
+	controllerConfig "github.com/numaproj-labs/numaplane/internal/controller/config"
+	gitShared "github.com/numaproj-labs/numaplane/internal/shared/git"
+)
+
+func CloneRepo(
+	ctx context.Context,
+	client k8sClient.Client,
+	gitSync *v1alpha1.GitSync,
+	namespace string,
+	globalConfig controllerConfig.GlobalConfig,
+) (*git.Repository, error) {
+	gitCredentials := gitShared.FindCredByUrl(gitSync.Spec.RepoUrl, globalConfig)
+	cloneOptions, err := gitShared.GetRepoCloneOptions(ctx, gitCredentials, client, namespace, gitSync.Spec.RepoUrl)
+	if err != nil {
+		return nil, fmt.Errorf("error getting  the  clone options: %v", err)
+	}
+	return cloneRepo(ctx, gitSync, cloneOptions)
+}
+
+// GetLatestManifests monitors a Git repository for changes. It fetches updates from the remote repository,
+// checks the latest commit hash against the stored hash in the GitSync object,
```

Comment:
Is this one a "todo" I guess?

Reply:
yes, the comment needs to be updated. Thanks!

Reply:
I believe the comment below about a "ticker" is an old comment as well, right?